addons/payment_adyen/controllers/main.py

```diff
@@ -1,37 +1,359 @@
-# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
 
+import base64
+import binascii
+import hashlib
+import hmac
 import json
 import logging
 import pprint
+
 import werkzeug
+from werkzeug import urls
 
-from odoo import http
+from odoo import _, http
+from odoo.exceptions import ValidationError
 from odoo.http import request
+from odoo.tools.pycompat import to_text
+
+from odoo.addons.payment import utils as payment_utils
+from odoo.addons.payment_adyen.const import CURRENCY_DECIMALS
 
 _logger = logging.getLogger(__name__)
 
 
 class AdyenController(http.Controller):
-    _return_url = '/payment/adyen/return/'
-
-    @http.route([
-        '/payment/adyen/return',
-    ], type='http', auth='public', csrf=False)
-    def adyen_return(self, **post):
-        _logger.info('Beginning Adyen form_feedback with post data %s', pprint.pformat(post))  # debug
-        if post.get('authResult') not in ['CANCELLED']:
-            request.env['payment.transaction'].sudo().form_feedback(post, 'adyen')
-        return werkzeug.utils.redirect('/payment/process')
-
-    @http.route([
-        '/payment/adyen/notification',
-    ], type='http', auth='public', methods=['POST'], csrf=False)
-    def adyen_notification(self, **post):
-        tx = post.get('merchantReference') and request.env['payment.transaction'].sudo().search([('reference', 'in', [post.get('merchantReference')])], limit=1)
-        if post.get('eventCode') in ['AUTHORISATION'] and tx:
-            states = (post.get('merchantReference'), post.get('success'), tx.state)
-            if (post.get('success') == 'true' and tx.state == 'done') or (post.get('success') == 'false' and tx.state in ['cancel', 'error']):
-                _logger.info('Notification from Adyen for the reference %s: received %s, state is %s', states)
+
+    @http.route('/payment/adyen/acquirer_state', type='json', auth='public')
+    def adyen_acquirer_state(self, acquirer_id):
+        """ Return the state of the acquirer.
+
+        :param int acquirer_id: The acquirer handling the transaction, as a `payment.acquirer` id
+        :return: The state of the acquirer: 'enabled' or 'test'
+        :rtype: str
+        """
+        acquirer_sudo = request.env['payment.acquirer'].sudo().browse(acquirer_id).exists()
```

Comment:
no verification for that?

Reply:
I don't feel that this is necessary. It's common for payment acquirer to have routes returning public data like the acquirer state. An access token check for this would be overkill IMO.